### PR TITLE
Xeno-based Hellhound

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

Refactors hellhounds to be based on xeno code. They share many similarities with runners and warriors now. They also regenerate health while standing, but get it much faster while resting. They do not need weeds to heal.

## Why It's Good For The Game

It's neat.

## Changelog

:cl:
refactor: Hellhounds are now based on xeno code, making them far less terrible to play as.
add: Forsaken and Yautja hives now never delay round end, Corrupted don't count as hostile xenos as long as they're allied to marines, as far as round-end is concerned.
/:cl:

### CODE REVIEW

**Review by Agameofscones — APPROVED** [review]
[Lester]: Looks like you mashed some poor feller's dog, Corporal.

[Vehicle Crewman]: That ain't no dog. It's a Hellhound, Lester. They shouldn't be out this far unless... oh shit.

Sprites approved.

File: code/modules/cm_preds/yaut_items.dm [hunk 1]
```diff
@@ -324,8 +324,8 @@
 		to_chat(M, SPAN_WARNING("You try to talk into the headset, but just get a horrible shrieking in your ears!"))
 		return
 
-	for(var/mob/living/carbon/hellhound/H in GLOB.player_list)
-		if(istype(H) && !H.stat)
+	for(var/mob/living/carbon/Xenomorph/Hellhound/H as anything in GLOB.hellhound_list)
```
Comment (stanalbatross) [inline code/modules/cm_preds/yaut_items.dm:327]: more specific var name

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -792,72 +792,38 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
-
-	var/list/hellhound_list = list()
-
-	for(var/mob/living/carbon/hellhound/A in GLOB.hellhound_list)
-		if(istype(A) && !A.client)
-			hellhound_list += A.real_name
-
-	if(hellhound_list.len == 0)
-		to_chat(usr, "<span style='color: red;'>There aren't any available Hellhounds.</span>")
+	var/mob/M = src
```
Comment (stanalbatross) [inline code/modules/mob/dead/observer/observer.dm:795]: var name again

File: code/modules/mob/living/carbon/xenomorph/castes/Hellhound.dm [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+			paw, \
+			sway, \
+			snow, \
+			whimper, \
+			grunt, \
+			rumble, \
+			howl, \
+			growl, \
+			stare, \
+			snift \
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/castes/Hellhound.dm:158]:
```suggestion
			sniff \
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Pull request merged**